source/slang/slang-ir.h

```diff
@@ -2239,6 +2240,8 @@ struct IRAnalysis
     IRDominatorTree* getDominatorTree();
 };
 
+struct DiagnosticSink;
```

Comment:
You may need to move this line up to the top where other classes are declared; around line 28~38.